include/boost/json/value.hpp

```diff
@@ -2832,6 +2832,65 @@ struct key_value_pair
 
 //----------------------------------------------------------
 
+#ifdef BOOST_JSON_DOCS
+
+/** Tuple-like element access.
+    
+    This overload permits the key and value
+    of a `key_value_pair` to be accessed
+    by index. For example:
+
+    @code 
+
+    key_value_pair kvp("num", 42);
+    
+    auto key = get<0>(kvp);
+    auto value = get<1>(kvp);
+
+    @endcode
+
+    In addition to calling the function directly,
+    this function allows `key_value_pair` objects
+    to initialize structured bindings:
+
+    @code
+
+    key_value_pair kvp("num", 42);
+
+    auto& [key, value] = kvp;
+
+    @endcode
+
+    The return type of this template
+    depends upon the value of `I`.
+    If `I` is 0, the return type is `const string_view`.
+    If `I` is 1, the return type is `value&` if
+    the argument is a non-const lvalue, `value&&` if 
+    the argument is a non-const rvalue, and 
+    `const value&` otherwise..
+    Any other value for `I` is ill-formed.
```

Comment:
```
Depending on the value of `I`, the return type will be:

@li `string_view const` if `I == 0`,

@li `value&` or `value const&` if `I == 1`, otherwise

@li the program is ill-formed.
```

Note that the docca toolchain has a preference for east-const, so we need to be consistent in user-facing program code and identifiers.